TC-62 Work Item list

LinetTorrico opened the pull request:
TC-62 Work Item list
1. Including work item list feature
2. Work Item List
![workitemlist](https://user-images.githubusercontent.com/29335751/41696427-193207e4-74e2-11e8-850f-2dee009f8171.PNG)

LinetTorrico committed:
TC-62 Work Item list

TC-62 Work Item list

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -23,6 +24,15 @@
+    @FindBy(id = "workItemMenu")
+    private WebElement workItemMenu;
+
+
```
please user format code
remove blank line

marcelogaray replied:
I see multiple blank lines, please fix it.

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -36,4 +46,16 @@ public RoleList clickOnRoleMenu() {
+        return new WorkItemList();
+    }
+
+
```
remove multiples black lines

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -23,6 +24,15 @@
+    //workItemSubMenu
+
     public void clickOnPersonnelMenu(){
         CommonEvents.clickButton(personnelMenu);
```
it should be returned a page object according to purpose

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/workItem/WorkItemList.java:
```diff
@@ -0,0 +1,23 @@
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class WorkItemList extends BasePage {
```
this class needs a constructure

marcelogaray commented on src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java:
```diff
@@ -25,23 +26,23 @@ public void ssiAHomePageIsLoaded() throws Throwable {
-    public void isEmployeeListPresent() throws Throwable{
+    public void isEmployeeListPresent() throws Throwable {
         boolean result = employeeList.isEmployeeListPresent();
         Assert.assertTrue(result);
```
when this assertion will fail then you will need to set an error message

marcelogaray commented on src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java:
```diff
@@ -60,6 +61,27 @@ public void roleListIsShowedInPage() throws Throwable {
+
+    @And("Clicking on Work Items menu on 'Header' page")
+    public void clickWorkItemsMenu() throws Throwable {
+        ssiaHome.clickOnWorkItemsMenu();
```
review your structure according to before observation

marcelogaray commented on src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/ssia/SSIAStepDefinitions.java:
```diff
@@ -60,6 +61,27 @@ public void roleListIsShowedInPage() throws Throwable {
+
+    @Then("^'Work Item List' page loads correctly$")
+    public void workItemListIsShowedInPage() throws Throwable {
+        Assert.assertTrue(workItemList.isWorkItemListPresent(), "Fail, Work Item list is not loaded");
```
workitmelist should be reviewed it.

marcelogaray reviewed (changes requested):
review observations

LinetTorrico committed:
Observations has been fixed

1. please user format code remove blank line (Fixed)
2. it should be returned a page object according to purpose (Fixed)
3. this class needs a constructure (Fixed)
4. when this assertion will fail then you will need to set an error message (Fixed)

LinetTorrico committed:
Fix errors space

LinetTorrico commented:
1. please user format code remove blank line (Fixed)
2. it should be returned a page object according to purpose (Fixed)
3. this class needs a constructure (Fixed)
4. when this assertion will fail then you will need to set an error message (Fixed)
![workitemlist](https://user-images.githubusercontent.com/29335751/41751541-e2a517e4-758f-11e8-8c4a-69421e184d65.PNG)

LinetTorrico committed:
Corrections

please user format code remove blank line (Fixed)
it should be returned a page object according to purpose (Fixed)
this class needs a constructure (Fixed)
when this assertion will fail then you will need to set an error message (Fixed)

LinetTorrico committed:
Clicking fixed

LinetTorrico committed:
On removed

LinetTorrico committed:
Removing on

Removing on

LinetTorrico commented:
On and Clicking has been removed

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -23,11 +25,19 @@
+
+
+    public void clickOnPersonnelMenu() {
         CommonEvents.clickButton(personnelMenu);
```
please rename this method clickPersonnelMenu

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -36,4 +46,14 @@ public RoleList clickOnRoleMenu() {
         return new RoleList();
     }
+
+    public void clickOnWorkItemsMenu() {
```
remove this method, it is not return an object.

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -36,4 +46,14 @@ public RoleList clickOnRoleMenu() {
+        CommonEvents.clickButton(workItemMenu);
+    }
+
+    public WorkItemList clickOnWorkItemMenu() {
```
please rename this method clickWorkItemMenu

marcelogaray reviewed (changes requested):
review observations

LinetTorrico committed:
Merge branch 'development' into TC-62WorkItemList

LinetTorrico committed:
Refactor menu Work Item and Fix Corrections

1. remove this method, it is not return an object. (Fixed)
2. please rename this method clickWorkItemMenu (Fixed)
3. Refactor Menu Work Item

LinetTorrico commented:
Corrections has been fixed 
![corrections 06-22-2018](https://user-images.githubusercontent.com/29335751/41792388-b8c9286a-7626-11e8-9151-5b14d4399a0d.png)

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -44,6 +52,12 @@ public RoleList clickOnRoleMenu() {
     public SafetyMenu clickSafetyMenu() {
         CommonEvents.clickButton(safetyMenu);
         return new SafetyMenu();
+
```
remove blank line

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/home/SSIAHome.java:
```diff
@@ -44,6 +52,12 @@ public RoleList clickOnRoleMenu() {
+    public WorkItemsMenu clickWorkItemsMenu() {
+        CommonEvents.clickButton(workItemMenu);
+        return new WorkItemsMenu();
+
```
remove blank line

marcelogaray commented on src/main/java/org/umssdiplo/automationv01/core/managepage/menuheader/workItemsMenu/WorkItemsMenu.java:
```diff
@@ -0,0 +1,19 @@
+        CommonEvents.clickButton(workItemMenu);
+        return new WorkItemList();
+    }
+
```
remove blank line

marcelogaray reviewed (changes requested):
review observations

marcelogaray commented:
Please fix conflicts and my observations
Send me a pull request once you fix it, I will close it in the meantime
